addons/purchase/static/src/toaster_button/toaster_button_widget.js

```diff
@@ -0,0 +1,35 @@
+/** @odoo-module */
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+
+const { Component } = owl;
+
+class ButtonWithNotification extends Component {
+    setup() {
+        this.orm = useService("orm");
+        this.notification = useService("notification");
+    }
+
+    get model() {
+        return this.props.record.resModel;
+    }
+
+    get id() {
+        return this.props.record.data.id;
+    }
```

Comment:
Creating two getters only to access them in a single place seems a little overkill. In particular, I would advise to avoid getters that do no work except accessing data elsewhere, as it just adds an extra layer of indirection that someone  reading the code needs to unwind.